iotlabmqtt/mqttcommon.py

```diff
@@ -20,6 +20,20 @@
 
 PAHO_VERSION = packaging.version.parse(paho.mqtt.__version__)
 
+class MQTTWrapMessage(object):
+    """MQTTMessage wrapper."""
+
+    def __init__(self, obj, reply_publisher):
+        """ Wrapper constructor. """
+        self._wrapped_obj = obj
+        self.reply_publisher = reply_publisher
+
+    def __getattr__(self, attr):
+        # see if this object has attr
+        if attr in self.__dict__:
```

Comment:
Could it be:
```python
if hasattr(self, attr)`:
```
instead ?

Reply:
I took this approach approach it's seems that `hasattr` for this specific action have a risk of ["infinite" recursion](https://nedbatchelder.com/blog/201010/surprising_getattr_recursion.html) (and minor comment [here](http://code.activestate.com/recipes/577555-object-wrapper-class/)).

Of course, I'm still not a full expert with python and its decorators, so if the general consensus is that `hasattr`  is more elegant and proper for this action. Then, let's do it :).